### ISSUE TITLE
With Raphnet 2 controller adapter both are controller 1 since release 0427

### ISSUE THREAD

**Issue by paulb-nl** [MiSTer-devel/Main_MiSTer#74]
I am using a Raphnet 2 controller Wii classic to USB adapter. Starting from Main release 20190427 both controllers will act as controller 1.

Log 0422
````
Input event: type=EV_KEY, code=316(0x13c), value=1, jnum=0, ID:289b:002c
FileLoad(open) File:/media/fat/config/kbd_289b_002c.map, error: -1.
PS2 code(make)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=0, jnum=1, ID:289b:002c
PS2 code(break)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=1, jnum=0, ID:289b:002c
PS2 code(make)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=0, jnum=2, ID:289b:002c
PS2 code(break)(ext) for core: 117(0x75)
````

Log 0427
````
Input event: type=EV_KEY, code=316(0x13c), value=1, jnum=0, ID:289b:002c
FileLoad(open) File:/media/fat/config/kbd_289b_002c.map, error: -1.
PS2 code(make)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=0, jnum=1, ID:289b:002c
PS2 code(break)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=1, jnum=1, ID:289b:002c
PS2 code(make)(ext) for core: 117(0x75)
Input event: type=EV_KEY, code=316(0x13c), value=0, jnum=1, ID:289b:002c
PS2 code(break)(ext) for core: 117(0x75)
`````

**Comment by sorgelig** [comment]
show me log when MiSTer opens input devices. So there should be important info. Probably this Raphnet adapter exposes uniq field and it's the same for all connected devices which basically violates the HID rule.

**Comment by paulb-nl** [comment]
Here are the full boot logs:
[mister_log_0422.txt](https://github.com/MiSTer-devel/Main_MiSTer/files/3144927/mister_log_0422.txt)
[mister_log_0427.txt](https://github.com/MiSTer-devel/Main_MiSTer/files/3144928/mister_log_0427.txt)

**Comment by paulb-nl** [comment]
Adapter usb info:

[raphnet_usbinfo.txt](https://github.com/MiSTer-devel/Main_MiSTer/files/3145414/raphnet_usbinfo.txt)

**Comment by sorgelig** [comment]
opened 0(-1): /dev/input/event1 (289b:002c) "105716" "raphnet technologies 2-player WUSBMote v2.1"
opened 1( 0): /dev/input/event0 (289b:002c) "105716" "raphnet technologies 2-player WUSBMote v2.1"

this is the root of problem.
It exposes uniq field to both controllers while it shoudn't. Thus new MiSTer treat them as a single controller. Uniq field is the only way MiSTer can combine multifunctional devices into a common game controller. For example DualShock exposes several input devices per gamepad. Same for Wiimote (after i've tweaked the driver) and other devices.

i can add workaround for 289b:002c

**Comment by paulb-nl** [comment]
Raphnet has various 2 controller adapters so these will all be affected.

It seems Dualshock and Wiimote have different names for their devices. What do you think about combining multifunctional gamepad only if it has different device names with same uniq?

**Comment by sorgelig** [comment]
I cannot relay on device names. It's not about only Dualshock and Wiimote. It's standard way to distinguish parts of the same device regardless the names.
if 289b is VID for all Raphnet devices, then i can disable uniq field for the whole this VID.

**Comment by sorgelig** [comment]
With native support of wiimote you can use it as a wireless adaptor for Classic controller. I see Aliexpress is selling other adapters for Wiimote to connect other wired gamepads. So, it looks better than Raphnet.

**Comment by paulb-nl** [comment]
Ok lets try VID only workaround.

**Comment by sorgelig** [comment]
https://devicehunt.com/view/type/usb/vendor/289B

**Comment by paulb-nl** [comment]
Fixed by a1d52030699b179f7c6f1b2f08d47193689c76da
